pipeline_tools/gcs_utils.py

```diff
@@ -0,0 +1,108 @@
+"""This module contains utility functions and classes to interact with Google Cloud Storage Service.
+"""
+import logging
+from google.cloud import storage
+import google.auth
+from io import BytesIO
+
+
+def get_filename_from_gs_link(link):
+    """Get the filename corresponding to a google_storage link.
+
+    :param str link: A string of google cloud storage link.
+    :return str: A string of filename.
+    """
+    return link.split('/')[-1]
+
+
+def parse_bucket_blob_from_gs_link(path):
+    """Utility to split a google storage path into bucket + blob name.
+
+    :param str path: A string of google cloud storage path (must have gs:// prefix)
+    :return str: A string of bucket name.
+    :return str: A string of blob name
+    """
+    if not path.startswith('gs://'):
+        raise ValueError('%s path is not a valid link')
+    parts = path.split('/', 3)
+    (prefix, _, bucket), blob = parts[:3], parts[3]
+
+    return bucket, blob
+
+
+def download_to_buffer(blob):
+    """Return a bytes file-like object readable by requests and REST APIs
+
+    :param google.cloud.storage.Blob blob: google storage blob
+    :return _io.BufferedIOBase: readable file object
+    """
+    string_buffer = BytesIO()
```

Comment:
I always try to name these `bytes_buffer` because `io.BufferedIOBase` doesn't clearly indicate that this is a bytes object you're getting.

Reply:
good idea